frontend/apps/reader/modules/readerstatus.lua

```diff
@@ -169,12 +169,8 @@ function ReaderStatus:openFileBrowser()
 end
 
 function ReaderStatus:onOpenNextDocumentInFolder()
-    local FileManager = require("apps/filemanager/filemanager")
-    if not FileManager.instance then
-        self.ui:showFileManager()
-    end
-    local next_file = FileManager.instance.file_chooser:getNextFile(self.document.file)
-    FileManager.instance:onClose()
+    local FileChooser = require("ui/widget/filechooser")
+    local next_file = FileChooser:getNextFile(self.document.file)
```

Comment:
Err, FileChooser isn't a singleton, we *do* need an instance of it for this to work?

Reply:
And if that actually worked, it means we have a much bigger problem of something storing stuff in the class's `item_table`... *somewhere* ?

Reply:
It's not a class's item_table, it's a local array that is destroyed after function is finished.

Reply:
Err, I'm definitely seeing that access `self.item_table` as populated in `init`, unless I'm missing something obvious?


https://github.com/koreader/koreader/blob/b949d07f4fbf7ffb0a145684f98f196bb0492aea/frontend/ui/widget/filechooser.lua#L233-L237

=>

https://github.com/koreader/koreader/blob/b949d07f4fbf7ffb0a145684f98f196bb0492aea/frontend/ui/widget/filechooser.lua#L561-L574

Reply:
Which is precisely why this was spinning up `FileManager` before ;).

Reply:
Please look at the diffs.

Reply:
Oh, see? Something obvious ;D.

My bad, not at the computer, so I completely missed the function name :D.

Reply:
And `show_finished` is a class member and used as such, so this is perfectly fine, cool! :}

Reply:
And this is also the only caller of `getNextFile`, so we don't have to worry about other call sites doing it differently.